eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/BlockchainWatchdog.scala

```diff
@@ -50,21 +51,29 @@ object BlockchainWatchdog {
   case class LatestHeaders(currentBlockCount: Long, blockHeaders: Set[BlockHeaderAt], source: String) extends Command
   private[watchdogs] case class WrappedCurrentBlockCount(currentBlockCount: Long) extends Command
   private case class CheckLatestHeaders(currentBlockCount: Long) extends Command
+  private case class NoBlockReceivedSince(lastBlockCount: Long) extends Command
   // @formatter:on
 
   /**
    * @param chainHash      chain we're interested in.
    * @param maxRandomDelay to avoid the herd effect whenever a block is created, we add a random delay before we query
    *                       secondary blockchain sources. This parameter specifies the maximum delay we'll allow.
    */
-  def apply(chainHash: ByteVector32, maxRandomDelay: FiniteDuration): Behavior[Command] = {
+  def apply(chainHash: ByteVector32, maxRandomDelay: FiniteDuration, blockTimeout: FiniteDuration = 15 minutes): Behavior[Command] = {
     Behaviors.setup { context =>
       context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[CurrentBlockCount](cbc => WrappedCurrentBlockCount(cbc.blockCount)))
       Behaviors.withTimers { timers =>
+        // We start a timer to check blockchain watchdogs regularly even when we don't receive any block.
+        timers.startSingleTimer(NoBlockReceivedTimer, NoBlockReceivedSince(0), blockTimeout)
         Behaviors.receiveMessage {
+          case NoBlockReceivedSince(lastBlockCount) =>
+            context.self ! CheckLatestHeaders(lastBlockCount)
+            timers.startSingleTimer(NoBlockReceivedTimer, NoBlockReceivedSince(lastBlockCount), blockTimeout)
```

Comment:
Doc:

> if a new timer with same key is started the previous is cancelled.

So `timers.startSingleTimer` effectively resets the timer. ✔️

Reply:
Yes, that's why I chose the overload with a `key` ;)